src/org/rascalmpl/library/Prelude.java

```diff
@@ -1495,47 +1495,25 @@ public IList readFileBytes(ISourceLocation sloc) {
 		return w.done();
 	}
 	
-    public IString readBase64(ISourceLocation sloc, IBool includePadding) {
-        final int BUFFER_SIZE = 3 * 512;
-        Base64.Encoder encoder = Base64.getEncoder();
-		if (!includePadding.getValue()) {
-			encoder = encoder.withoutPadding();
+	public IString readBase64(ISourceLocation sloc, IBool includePadding) {
+		StringBuilder result = new StringBuilder();
+		try (var source = REGISTRY.getInputStream(sloc)) {
+			StreamingBase64.encode(source, result, includePadding.getValue());
 		}
-        
-        try  (BufferedInputStream in = new BufferedInputStream(REGISTRY.getInputStream(sloc), BUFFER_SIZE); ) {
-            StringBuilder result = new StringBuilder();
-            byte[] chunk = new byte[BUFFER_SIZE];
-            int len = 0;
-            
-            // read multiples of 3 until not possible anymore
-            while ( (len = in.read(chunk)) == BUFFER_SIZE ) {
-				result.append(new String(encoder.encode(chunk), StandardCharsets.ISO_8859_1));
-            }
```

Comment:
this was a bug, since inputstreams are not required to always fill the full buffer, they only have to read at least one byte.